### PR TITLE
Setting log group retention for lambda functions.

### PR DESCRIPTION
**Pull request by ben851**
# Summary | Résumé

Following up on Steve's QA of a few log groups still having incorrect retention periods, I determined that the three lambdas were buried in the module. I've patched the module source code and updated our modules to use the latest.

In addition to setting the retention period, the module will create a log query to provide basic stats on the lambda

The BatchSaving log group doesn't appear to be in terraform, so I added it in. This means we will need to import it into staging and prod

# Deployment instructions:
The following command must be run manually before merging (but after approval)

```
env/<env>/elasticache

terragrunt import -var-file <env.tfvars> aws_cloudwatch_log_group.redis-batch-saving BatchSaving
```

# Test instructions | Instructions pour tester la modification

Terraform apply works
Verify log retention period is correct for these:
[Batch Saving](https://ca-central-1.console.aws.amazon.com/cloudwatch/home?region=ca-central-1#logsV2:log-groups/log-group/BatchSaving) 
[/aws/lambda/ses_to_sqs_email_callbacks](https://ca-central-1.console.aws.amazon.com/cloudwatch/home?region=ca-central-1#logsV2:log-groups/log-group/$252Faws$252Flambda$252Fses_to_sqs_email_callbacks) 
[/aws/lambda/sns_to_sqs_sms_callbacks](https://ca-central-1.console.aws.amazon.com/cloudwatch/home?region=ca-central-1#logsV2:log-groups/log-group/$252Faws$252Flambda$252Fsns_to_sqs_sms_callbacks)
[/aws/lambda/ses-receiving-emails](https://ca-central-1.console.aws.amazon.com/cloudwatch/home?region=ca-central-1#logsV2:log-groups/log-group/$252Faws$252Flambda$252Fses-receiving-emails)

### CODE REVIEW

File: aws/sns_to_sqs_sms_callbacks/lambda.tf [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-  memory                 = 1024
+  source                     = "github.com/cds-snc/terraform-modules//lambda?ref=v7.3.3"
+  name                       = "sns_to_sqs_sms_callbacks"
+  billing_tag_value          = var.billing_tag_value
+  ecr_arn                    = var.sns_to_sqs_sms_callbacks_ecr_arn
+  enable_lambda_insights     = true
+  image_uri                  = "${var.sns_to_sqs_sms_callbacks_ecr_repository_url}:${var.sns_to_sqs_sms_callbacks_docker_tag}"
+  timeout                    = 60
+  memory                     = 1024
+  log_group_retention_period = var.env == "production" ? 0 : 365
```
Comment (jimleroyer) [inline aws/sns_to_sqs_sms_callbacks/lambda.tf:10]: This log group specifically will contain the phone numbers of our users. So we might have to downgrade to 7 days if I get our data retention rights when applied to logs.

File: aws/ses_receiving_emails/lambda.tf [hunk 2]
```diff
@@ -4,14 +4,15 @@ module "ses_receiving_emails" {
...
-  source                 = "github.com/cds-snc/terraform-modules//lambda?ref=v4.0.3"
-  name                   = "ses_receiving_emails"
-  billing_tag_value      = var.billing_tag_value
-  ecr_arn                = var.ses_receiving_emails_ecr_arn
-  enable_lambda_insights = true
-  image_uri              = "${var.ses_receiving_emails_ecr_repository_url}:${var.ses_receiving_emails_docker_tag}"
-  timeout                = 60
-  memory                 = 1024
+  source                     = "github.com/cds-snc/terraform-modules//lambda?ref=v7.3.3"
+  name                       = "ses_receiving_emails"
```
Comment (jimleroyer) [inline aws/ses_receiving_emails/lambda.tf:8]: I don't see a log group with that name in AWS at the moment. Did we drop this lambda? or it's not logging?

Comment (ben851) [reply aws/ses_receiving_emails/lambda.tf:8]: Good question... the log group is there, why it's not logging is something we should investigate!

Comment (ben851) [reply aws/ses_receiving_emails/lambda.tf:8]: [Card created here](https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/236)

**Review by jimleroyer — APPROVED** [review]
LGTM. Do we want to revisit this later to enable longer logs retention with redacted values?

**Pull request merged**